aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -167,30 +165,50 @@ module aptos_framework::coin {
         supply: Option<OptionalAggregator>,
     }
 
-    /// Event emitted when some amount of a coin is deposited into an account.
-    struct DepositEvent has drop, store {
+
+    #[event]
+    /// Module event emitted when some amount of a coin is deposited into an account.
+    struct CoinDeposit has drop, store {
+        coin_type: String,
+        account: address,
         amount: u64,
     }
 
     #[event]
-    /// Module event emitted when some amount of a coin is deposited into an account.
-    struct Deposit<phantom CoinType> has drop, store {
+    /// Module event emitted when some amount of a coin is withdrawn from an account.
+    struct CoinWithdraw has drop, store {
+        coin_type: String,
         account: address,
         amount: u64,
     }
 
-    /// Event emitted when some amount of a coin is withdrawn from an account.
-    struct WithdrawEvent has drop, store {
+    // DEPRECATED, NEVER USED
+    #[deprecated]
+    #[event]
+    struct Deposit<phantom CoinType> has drop, store {
+        account: address,
         amount: u64,
     }
 
+    // DEPRECATED, NEVER USED
+    #[deprecated]
     #[event]
-    /// Module event emitted when some amount of a coin is withdrawn from an account.
     struct Withdraw<phantom CoinType> has drop, store {
         account: address,
         amount: u64,
     }
 
+    /// Event emitted when some amount of a coin is deposited into an account.
+    struct DepositEvent has drop, store {
+        amount: u64,
+    }
+
+    /// Event emitted when some amount of a coin is withdrawn from an account.
+    struct WithdrawEvent has drop, store {
+        amount: u64,
+    }
+
+
```

Comment:
Why do we need these?

Reply:
please check. they are moved here but not new code...